src/Anthropic.Client.Tests/Services/Messages/MessageServiceTest.cs

```diff
@@ -38,6 +42,213 @@ public async Task CreateStreaming_Works()
+        var messagesServiceMock = new Mock<IMessageService>();
+        static async IAsyncEnumerable<RawMessageStreamEvent> GetTestValues()
+        {
+            yield return new(new RawMessageStartEvent(GenerateStartMessage()));
+            yield return new(new RawMessageStopEvent());
+            await Task.CompletedTask;
+        }
+        // act
+        messagesServiceMock.Setup(e => e.CreateStreaming(It.IsAny<MessageCreateParams>())).Returns(GetTestValues);
+        var stream = await messagesServiceMock.Object.CreateStreaming(
+            new()
+            {
+                MaxTokens = 1024,
+                Messages = [new() { Content = new(""), Role = Role.User }],
+                Model = Model.Claude3_7SonnetLatest,
+            }
+        ).Aggregate();
+
+        Assert.NotNull(stream);
+        Assert.Null(stream.Text);
+        Assert.Null(stream.Thinking);
+        Assert.NotNull(stream.Citations);
+        Assert.Empty(stream.Citations);
+        Assert.Null(stream.Thinking);
+    }
+
+    [Fact]
+    public async Task CreateStreamingAggregation_HandlesNoEndMessageInterrupt()
+    {
+        // arrange
+        var messagesServiceMock = new Mock<IMessageService>();
+        static async IAsyncEnumerable<RawMessageStreamEvent> GetTestValues()
+        {
+            yield return new(new RawMessageStartEvent(GenerateStartMessage()));
+            await Task.CompletedTask;
+        }
+        // act
+        messagesServiceMock.Setup(e => e.CreateStreaming(It.IsAny<MessageCreateParams>())).Returns(GetTestValues);
+
+        // assert
+        await Assert.ThrowsAsync<InvalidOperationException>(async () => await messagesServiceMock.Object.CreateStreaming(
+            new()
+            {
+                MaxTokens = 1024,
+                Messages = [new() { Content = new(""), Role = Role.User }],
+                Model = Model.Claude3_7SonnetLatest,
+            }
+        ).Aggregate());
+    }
+
+    [Fact]
+    public async Task CreateStreamingAggregation_WorksNoContent_RawContentBlockStartEvent()
+    {
+        // arrange
+        var messagesServiceMock = new Mock<IMessageService>();
+        static async IAsyncEnumerable<RawMessageStreamEvent> GetTestValues()
+        {
+            yield return new(new RawContentBlockStartEvent()
+            {
+                Index = 0,
+                ContentBlock = null
+            });
+            yield return new(new RawContentBlockStopEvent()
+            {
+                Index = 1,
+            });
+            await Task.CompletedTask;
+        }
+        messagesServiceMock.Setup(e => e.CreateStreaming(It.IsAny<MessageCreateParams>())).Returns(GetTestValues);
+        // act
+        var stream = await messagesServiceMock.Object.CreateStreaming(
+            new()
+            {
+                MaxTokens = 1024,
+                Messages = [new() { Content = new(""), Role = Role.User }],
+                Model = Model.Claude3_7SonnetLatest,
+            }

```

Comment:
Might be nice to extract the params to a static if it's the same for every one of these tests. Makes it easier to tell what matters about the test vs what's just other data